pkg/operators/machine-api-operator.go

```diff
@@ -65,6 +66,93 @@ var _ = Describe("[Feature:Operators] Machine API operator deployment should", f
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(framework.IsMutatingWebhookConfigurationMatched(client)).To(BeTrue())
+	})
+
+	It("reconcile validating webhook configuration", func() {
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(framework.IsValidatingWebhookConfigurationMatched(client)).To(BeTrue())
+	})
+
+	It("recover after validating webhook configuration deletion", func() {
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(framework.DeleteValidatingWebhookConfiguration(client, framework.DefaultValidatingWebhookConfiguration)).To(Succeed())
+
+		Expect(framework.IsValidatingWebhookConfigurationMatched(client)).To(BeTrue())
+	})
+
+	It("recover after mutating webhook configuration deletion", func() {
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(framework.DeleteMutatingWebhookConfiguration(client, framework.DefaultMutatingWebhookConfiguration)).To(Succeed())
+
+		Expect(framework.IsMutatingWebhookConfigurationMatched(client)).To(BeTrue())
+	})
+
+	It("maintains spec after mutating webhook configuration change and preserve caBundle", func() {
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		initial, err := framework.GetMutatingWebhookConfiguration(client, framework.DefaultMutatingWebhookConfiguration.Name)
+		Expect(initial).ToNot(BeNil())
+		Expect(err).To(Succeed())
+
+		toUpdate := initial.DeepCopy()
+		for _, webhook := range toUpdate.Webhooks {
+			webhook.ClientConfig.CABundle = []byte("test")
+			webhook.AdmissionReviewVersions = append(webhook.AdmissionReviewVersions, "v1")
+		}
+
+		Expect(framework.UpdateMutatingWebhookConfiguration(client, toUpdate)).To(Succeed())
+
+		Expect(framework.IsMutatingWebhookConfigurationMatched(client)).To(BeTrue())
+
+		updated, err := framework.GetMutatingWebhookConfiguration(client, framework.DefaultMutatingWebhookConfiguration.Name)
+		Expect(updated).ToNot(BeNil())
+		Expect(err).To(Succeed())
+
+		for i, webhook := range updated.Webhooks {
+			Expect(webhook.ClientConfig.CABundle).To(Equal(initial.Webhooks[i].ClientConfig.CABundle))
+		}
+	})
+
+	It("maintains spec after validating webhook configuration change and preserve caBundle", func() {
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		initial, err := framework.GetValidatingWebhookConfiguration(client, framework.DefaultValidatingWebhookConfiguration.Name)
+		Expect(initial).ToNot(BeNil())
+		Expect(err).To(Succeed())
+
+		toUpdate := initial.DeepCopy()
+		for _, webhook := range toUpdate.Webhooks {
+			webhook.ClientConfig.CABundle = []byte("test")
+			webhook.AdmissionReviewVersions = append(webhook.AdmissionReviewVersions, "v1")
+		}
+
+		Expect(framework.UpdateValidatingWebhookConfiguration(client, toUpdate)).To(Succeed())
+
+		Expect(framework.IsValidatingWebhookConfigurationMatched(client)).To(BeTrue())
```

Comment:
Surely after updating it can never match the default again, because there will be no CA Bundle in there?

Reply:
It's the place where `DeepDerivative` would still fit, as it will ignore differences with empty field on the default resource. So it will still match :)